ra_aid/chat_models/deepseek_chat.py

```
@@ -0,0 +1,52 @@
+from langchain_core.callbacks import CallbackManagerForLLMRun
+from langchain_core.messages import BaseMessage
+from langchain_core.outputs import ChatResult
+from langchain_openai import ChatOpenAI
+from typing import Any, List, Optional, Dict
+
+
+# Docs: https://api-docs.deepseek.com/guides/reasoning_model
+class ChatDeepseekReasoner(ChatOpenAI):
+    """ChatDeepseekReasoner with custom overrides for R1/reasoner models."""
+
+    def __init__(self, *args, **kwargs):
```

Ruff (0.8.2)

Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)